src/main/java/evo/ui/Ui.java

```diff
@@ -0,0 +1,87 @@
+package evo.ui;
```

Comment:
I think you have done a great job, your code is very clear and it didn't take me much time to trace through your code👍.